HealthyPlanners/Urban-Sherpa
# Implement placeholder

Please note this PR touches only the App.jsx file, the UserSection.jsx file, the MapPlaceholder folder, and the Sphere Loader (both in client/components/common/experimental). All other files listed are part of earlier PRs

**Tasks Completed**
Sphere Loader Created and integrated into existing code
Map Placeholder Created and integrated into existing code
Common Suspend element for rendering placeholder created and integrated into existing code

**To Test**
Click either button (Sign up or Log in) to get to the user page. A placeholder shadow of the map should render for at least one second and then be replaced by the Google Map returned from the Google Maps API. After entering and selecting an address in the search bar and clicking get health, you should also see the place holder appear for a minimum of one second AND you should also briefly see the walk score div (in the center between the two wheel loaders) replaced with three jumping dots. Ultimately we should choose between dot loader and wheel loaders for a uniform look.

**The Files**
**App.jsx**
Cosmetic changes to avoid linting errors

**User Section**
Virtual complete rewrite to avoid linting errors and strip out old code. Also changed extension from JS to JSX. Add functionality of rendering map placeholder instead of Map when necessary.

**Suspend.jsx**
Newly Created
A new reusable component that renders a placeholder in place of a passed in component. 

**MapPlaceholder.jsx && MapPlaceholder.css**
Newly Created
The shadow component rendered in place of the map when API calls are being made

**SphereLoader.jsx && SphereLoader.css**
Newly Created
Three jumping dots that will repeat indefinitely. Meant to be rendered in place of components when API calls are being made
